commandline/micronucleus.c

```diff
@@ -155,8 +158,10 @@ int main(int argc, char **argv) {
 
   setProgressData("waiting", 1);
   if (dump_progress) printProgress(0.5);
-  printf("> Please plug in the device ... \n");
-  printf("> Press CTRL+C to terminate the program.\n");
+  fprintf(stderr, "> Please plug in or reset the device");
```

Comment:
Please put a comment here why using stderr. 

Resetting is no option in every case, on Windows I get 
`Assertion failed: res >= 6, file library/micronucleus_lib.c, line 69`
if device is already attached.

Reply:
> Please put a comment here why using stderr.

Thanks!

Reply:
> Please put a comment here why using stderr.
> 

Thanks for the quick response, comment is added.

> Resetting is no option in every case, on Windows I get
> `Assertion failed: res >= 6, file library/micronucleus_lib.c, line 69`
> if device is already attached.

Hmm, how would you recommend handling that?  I always though the message was a little misleading since I haven't found it necessary to unplug the device, but I don't have access to a windows machine to test that on.  Anyway, the launcher I'm hoping to replace also doesn't have the reset messaging, so it wouldn't cause a regression to remove that.

Reply:
I did a bit of research.
The message means, that a control request to the attached USB device gets no answer (of at least 6 bytes).
This happens if the usb device is not listening, but did not disconnect from the USB bus, which is a desirable behavior, since otherwise you get that nasty error in device manager.
See https://github.com/ArminJo/micronucleus-firmware/issues/2

At least we should change the line 
`assert(res >= 6);`
to 
```
if (res<6) {
fprintf(stderr, "Bus %s device %s: %s seems to be inactive. Please unplug or reset device.\n", bus->dirname, dev->filename, strerror(errno));
return NULL;
}
```
But I do not know  if bus->dirname etc. are the right fields for displaying the device manager info libusb-win32 devices and Digispark Bootloader. maybe you know better which fields to include in the error message.

![2020-12-06 00_12_40-Geräte-Manager](https://user-images.githubusercontent.com/3881884/101267361-070eeb00-3758-11eb-88ed-6821eeea90e6.png)

Reply:
Unfortunately I don't know enough about windows to add much there.  Using assert to catch a run time error does seem wrong, so I think your suggestion would be an improvement.
I wonder why Windows behaves so much differently than Linux?  On Linux if it's already connected to USB there's no error, if just gives the message to plug in as if it doesn't see it, then once it's reset the programming happens immediately.  Maybe there's same way to get similar behavior on Windows?

Reply:
Maybe tomorrow I can test it with Linux

Reply:
OK, I can remove that bit for now to avoid conflating too much into one PR.  It seems like there's some sense of urgency around releasing an updated core, so it probably makes sense to update the messaging in tandem with the Windows improvements.